File: fragmentation-core-androidx/src/main/java/me/yokeyword/fragmentation/anim/FragmentAnimator.java
package me.yokeyword.fragmentation.anim;

import android.os.Parcel;
import android.os.Parcelable;

import androidx.annotation.AnimRes;

/**
 * Fragment动画实体类
 * Created by YoKeyword on 16/2/4.
 */
@SuppressWarnings({"WeakerAccess", "unused"})
public class FragmentAnimator implements Parcelable {
    @AnimRes
    protected int mEnter;
    @AnimRes
    protected int mExit;
    @AnimRes
    protected int mPopEnter;
    @AnimRes
    protected int mPopExit;

    public FragmentAnimator() {
    }

    public FragmentAnimator(int enter, int exit) {
        this.mEnter = enter;
        this.mExit = exit;
    }

    public FragmentAnimator(int enter, int exit, int popEnter, int popExit) {
        this.mEnter = enter;
        this.mExit = exit;
        this.mPopEnter = popEnter;
        this.mPopExit = popExit;
    }

    public FragmentAnimator copy() {
        return new FragmentAnimator(getEnter(), getExit(), getPopEnter(), getPopExit());
    }

    public int getEnter() {
        return mEnter;
    }

    public FragmentAnimator setEnter(int enter) {
        this.mEnter = enter;
        return this;
    }

    public int getExit() {
        return mExit;
    }

    public FragmentAnimator setExit(int exit) {
        this.mExit = exit;
        return this;
    }

    public int getPopEnter() {
        return mPopEnter;
    }

    public FragmentAnimator setPopEnter(int popEnter) {
        this.mPopEnter = popEnter;
        return this;
    }

    public int getPopExit() {
        return mPopExit;
    }

    public FragmentAnimator setPopExit(int popExit) {
        this.mPopExit = popExit;
        return this;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(this.mEnter);
        dest.writeInt(this.mExit);
        dest.writeInt(this.mPopEnter);
        dest.writeInt(this.mPopExit);
    }

    protected FragmentAnimator(Parcel in) {
        this.mEnter = in.readInt();
        this.mExit = in.readInt();
        this.mPopEnter = in.readInt();
        this.mPopExit = in.readInt();
    }

    public static final Creator<FragmentAnimator> CREATOR = new Creator<FragmentAnimator>() {
        @Override
        public FragmentAnimator createFromParcel(Parcel source) {
            return new FragmentAnimator(source);
        }

        @Override
        public FragmentAnimator[] newArray(int size) {
            return new FragmentAnimator[size];
        }
    };
}

File: fragmentation-core-androidx/src/main/java/me/yokeyword/fragmentation/anim/FragmentAnimator.kt
package me.yokeyword.fragmentation.anim

import android.os.Parcel
import android.os.Parcelable

import androidx.annotation.AnimRes

/**
 * Fragment动画实体类
 * Created by YoKeyword on 16/2/4.
 */
@Suppress("unused")
open class FragmentAnimator : Parcelable {
    @AnimRes
    protected var enterAnim: Int = 0

    @AnimRes
    protected var exitAnim: Int = 0

    @AnimRes
    protected var popEnterAnim: Int = 0

    @AnimRes
    protected var popExitAnim: Int = 0

    constructor()

    constructor(enter: Int, exit: Int) {
        this.enterAnim = enter
        this.exitAnim = exit
    }

    constructor(enter: Int, exit: Int, popEnter: Int, popExit: Int) {
        this.enterAnim = enter
        this.exitAnim = exit
        this.popEnterAnim = popEnter
        this.popExitAnim = popExit
    }

    fun copy(): FragmentAnimator {
        return FragmentAnimator(getEnter(), getExit(), getPopEnter(), getPopExit())
    }

    fun getEnter(): Int {
        return enterAnim
    }

    fun setEnter(enter: Int): FragmentAnimator {
        this.enterAnim = enter
        return this
    }

    fun getExit(): Int {
        return exitAnim
    }

    fun setExit(exit: Int): FragmentAnimator {
        this.exitAnim = exit
        return this
    }

    fun getPopEnter(): Int {
        return popEnterAnim
    }

    fun setPopEnter(popEnter: Int): FragmentAnimator {
        this.popEnterAnim = popEnter
        return this
    }

    fun getPopExit(): Int {
        return popExitAnim
    }

    fun setPopExit(popExit: Int): FragmentAnimator {
        this.popExitAnim = popExit
        return this
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(this.enterAnim)
        dest.writeInt(this.exitAnim)
        dest.writeInt(this.popEnterAnim)
        dest.writeInt(this.popExitAnim)
    }

    protected constructor(`in`: Parcel) {
        this.enterAnim = `in`.readInt()
        this.exitAnim = `in`.readInt()
        this.popEnterAnim = `in`.readInt()
        this.popExitAnim = `in`.readInt()
    }

    companion object {

        @JvmField
        val CREATOR: Parcelable.Creator<FragmentAnimator> = object : Parcelable.Creator<FragmentAnimator> {
            override fun createFromParcel(source: Parcel): FragmentAnimator {
                return FragmentAnimator(source)
            }

            override fun newArray(size: Int): Array<FragmentAnimator?> {
                return arrayOfNulls(size)
            }
        }
    }
}
